packages/core/core/src/services/document-service/index.ts

```diff
@@ -131,7 +131,10 @@ const createDocumentService = ({
 
     const model = strapi.getModel(uid) as Shared.ContentTypes[Common.UID.ContentType];
 
-    const validData = await entityValidator.validateEntityCreation(model, data, { isDraft: true });
+    const validData = await entityValidator.validateEntityCreation(model, data, {
+      isDraft: true,
```

Comment:
I just realized, what about publishing a document? Shouldn't it be validating a publsihed version entity creation?

Reply:
Could you elaborate?

Reply:
when publishing a draft version, we need the validations to apply to the published versions , for example:

Given a content type with a unique field named code

Doc1-Draft  { code: 1 }

Doc2-Draft { code: 2 }
Doc2-Published { code :  1}

If you were to publish Doc1-Draft , shouldn't you get an error saying code is a unique attribute and that value already exist in Doc2-Published?

Side question, we want that behaviour right?

Reply:
Yep thats the right behaviour and I think it's what we have atm, see video. Is there something I've missed?


https://github.com/strapi/strapi/assets/48524071/ddf2a04b-dc68-43f7-8bcf-02e559cd64f9

_I should have just spoken over this video sorry, hope it's clear!_

Reply:
oh :O , then I wonder how i tworks. Why do we have an "isDraft" attribute that that is always true? that parameter is never sent as false. So I wonder at which point we do this validation for published versions.

Reply:
Publishing comes through the document-engine.clone here - https://github.com/strapi/strapi/blob/81fd684402ab62b66eba947c1af8a71a333b3621/packages/core/core/src/services/document-service/document-engine.ts#L227

where isDraft is set. We only default to true in the create & update e.g. https://github.com/strapi/strapi/blob/81fd684402ab62b66eba947c1af8a71a333b3621/packages/core/core/src/services/document-service/document-engine.ts#L177 which I think is correct behaviour?